astrocloud release 1.4.0 (#602)
astronomer/docs: astro/cli-reference/astrocloud-deployment-variable-create.md

---

## Description

Create Deployment-level environment variables by supplying either a key and value or an environment file with a list of keys and values.

## Usage

```sh
astrocloud deployment variable create
```

## Options

| Option                         | Description                                                                            | Possible Values                                                                |
| ------------------------------ | -------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------ |
| `-d`,`--deployment-id`           |       The Deployment to create variable(s) for                             | Any valid Deployment ID |
| `-e`,`--env`                  | Location of file to load environment variables from (default is `.env` in your current directory)                                                                  | Any valid filepath       |
| `-k`,`--key`             | The key for a new variable                                                  | Any string |

> | `-l`,`--load`    | Create environment variables based on an `.env` file             |`` |

Reviewer: @sunkickr In the example that the CLI outputs, we use `--file` instead of `--load`:
```
$ astro deployment variable create <deployment-id> --file --env .my-deployment-env
```
Is this a typo? Which option is the correct one for loading from an .env file?

Author: yes this is a typo. It was file at first but we changed it to load

| `-s`,`--secret`    | Set the new environment variable as a secret          |`` |
| `-u`,`--update`    | Update an existing variable with the specified key         |Any string |
| `-v`,`--value`    | Value for the new variable          |`` |
| `-w`,`--workspace-id`          | List environment variables for a Deployment outside of your current Workspace           | Any valid Workspace ID                                                         |

## Examples

```sh
# Create a new secret environment variable
$ astrocloud deployment variable create cl03oiq7d80402nwn7fsl3dmv --key AIRFLOW_VAR_MY_VAR --value my-secret-value --secret

> # Set multiple variables at once using an .env file

Reviewer 2: ```suggestion
# Create multiple environment variables for a Deployment at once by loading them from a .env file
```

> $ astrocloud deployment variable create cl03oiq7d80402nwn7fsl3dmv --load --env /users/documents/my-astro-project/.my-env

Reviewer: In the example generated by the CLI, we use `.my-deployment-env`. Is that the correct format, or should it be `my-deployment.env`? Can you name it anything as long as it starts with a `.`?

Author: you can name the file anything you like for the commands to work. For a file to work locally as an env file it must be written as `<file-name>.env` the default env file is usually named `.env`.

> # Update an existing variable

Reviewer 2: ```suggestion
# Update the value of an existing environment variable
```

| `-l`,`--load`    | Load new environment variables from a file. Specify this flag if the variables you want to create are in that file. This is an alternative to creating an environment variable by specifying `--key` and `--value`             |`` |
| `-s`,`--secret`    | Set the value of the new environment variable as secret      |`` |
| `-u`,`--update`    | Update the value of an existing environment variable. Make sure to specify the `--key` to which this new value will be applied          | Any string |
| `-v`,`--value`    | The environment variable value          |`` |
| `-w`,`--workspace-id`          | Create or update an environment variable for a Deployment that is not in your current Workspace. If this is not specified, your current Workspace is assumed           | Any valid Workspace ID                                                         |

## Examples

```sh
# Create a new secret environment variable
$ astrocloud deployment variable create cl03oiq7d80402nwn7fsl3dmv --key AIRFLOW__SECRETS__BACKEND_KWARGS --value <my-secret-value> --secret

# Create multiple environment variables for a Deployment at once by loading them from a .env file

> $ astrocloud deployment variable create cl03oiq7d80402nwn7fsl3dmv --load --env /users/documents/my-astro-project/.env

Author: ```suggestion
$ astrocloud deployment variable create cl03oiq7d80402nwn7fsl3dmv --load --env dev.env
```
